components/Themes.js

```diff
@@ -0,0 +1,361 @@
+        }
+
+        .name-field {
+          padding: 8px;
+        }
+
+        .theme-select {
+          width: 100%;
+        }
+
+        .theme-settings {
+          display: flex;
+          flex-direction: column;
+          width: 224px;
+        }
+
+        .theme-colors {
+          display: flex;
+          flex-wrap: wrap;
+          border-top: 2px solid ${COLORS.SECONDARY};
+          text-transform: capitalize;
+        }
+
+        .theme-colors .field {
+          width: 50%;
+        }
+
+        .theme-colors .field:nth-child(odd) {
+          border-right: 1px solid ${COLORS.SECONDARY};
+        }
+
+        .color-square {
+          border-radius: 3px;
+          padding: 12px;
+        }
+      `}
+    </style>
+  </Popout>
+)
+
+const ThemeItem = ({ children, item, isSelected, onClick }) => (
+  <div className="theme-item">
+    {children}
+    {item.custom && !isSelected && (
+      <div className="icon" onClick={onClick(item.id)}>
+        <RemoveIcon color={COLORS.SECONDARY} />
+      </div>
+    )}
+    <style jsx>
+      {`
+        .theme-item {
+          display: flex;
+          flex: 1;
+          justify-content: ${item.id === 'create' ? 'center' : 'space-between'};
+          align-items: center;
+        }
+
+        .icon {
+          display: flex;
+        }
+      `}
+    </style>
+  </div>
+)
+
+const themeIcon = <ThemeIcon />
+
+class Themes extends React.PureComponent {
+  constructor(props) {
+    super(props)
+
+    const storedThemes = getThemes(localStorage) || []
+    const themes = [...storedThemes, ...THEMES]
+    const highlights = themes.find(({ id }) => id === props.theme).highlights
+
+    this.state = {
+      preset: props.theme,
+      themes,
+      highlights,
+      name: '',
+      selected: null
+    }
+  }
+
+  applyPreset = preset =>
+    this.setState(({ themes }) => ({
+      preset,
+      highlights: themes.find(({ id }) => id === preset).highlights
+    }))
+
+  handleDropdown = ({ id }) => {
+    if (id === 'create') {
+      this.props.toggleVisibility()
+    } else {
+      this.props.updateTheme(id)
+    }
+  }
+
+  updateName = ({ target: { value: name } }) => this.setState({ name })
+
+  selectHighlight = highlight => () =>
+    this.setState(({ selected }) => ({
+      selected: selected === highlight ? null : highlight
+    }))
+
+  updateHighlight = ({ rgb }) =>
+    this.setState({
+      highlights: {
+        ...this.state.highlights,
+        [this.state.selected]: stringifyRGBA(rgb)
+      }
+    })
+
+  removeTheme = id => () => {
+    const { themes } = this.state
+
+    const newThemes = themes.filter(t => t.id !== id)
+
+    saveThemes(localStorage, newThemes.filter(({ custom }) => custom))
+
+    if (this.props.theme === id) {
+      this.props.updateTheme(DEFAULT_THEME.id)
+    } else {
+      this.setState({ themes: newThemes })
+    }
+  }
+
+  createTheme = () => {
+    const { themes, name, highlights } = this.state
+
+    const id = generateId()
```

Comment:
The `name` should be the `id` in this case

Reply:
I didn't use the name to avoid the edge case where the name already exists.

Reply:
Yeah I don't think we should allow creating custom themes with the same name 🙂